src/engine/move_handling.h

```diff
@@ -202,7 +202,7 @@ constexpr BitBoard performMove(const BitBoard& board, const movegen::Move& move,
         performPromotionMove(newBoard, move, hash);
     } else {
         if (move.isCapture()) {
-            if (const auto victim = board.getPieceAtSquare(move.toSquare())) {
+            if (const auto victim = (newBoard.player == PlayerWhite) ? board.getPieceAtSquare<PlayerWhite>(move.toSquare()) : board.getPieceAtSquare<PlayerBlack>(move.toSquare())) {
```

Comment:
This is the only exception to my above comment, because there's one other check for the colour in this function.

However I tried rewriting so the check was the outer of the nested conditionals, and it ended up with quite a few lines of duplication that IMO isn't worth the possible benefits at the current time.